components/app-core/backend/cnsi.go

```diff
@@ -262,19 +262,33 @@ func (p *portalProxy) GetCNSIRecord(guid string) (interfaces.CNSIRecord, error)
 	return rec, nil
 }
 
-func (p *portalProxy) cnsiRecordExists(endpoint string) bool {
-	log.Debug("cnsiRecordExists")
+func (p *portalProxy) GetCNSIRecordByEndpoint(endpoint string) (interfaces.CNSIRecord, error) {
+	log.Debug("GetCNSIRecordByEndpoint")
+	var rec interfaces.CNSIRecord
+
 	cnsiRepo, err := cnsis.NewPostgresCNSIRepository(p.DatabaseConnectionPool)
 	if err != nil {
-		return false
+		return rec, err
 	}
 
-	_, err = cnsiRepo.FindByAPIEndpoint(endpoint)
+	rec, err = cnsiRepo.FindByAPIEndpoint(endpoint)
 	if err != nil {
-		return false
+		return rec, err
 	}
 
-	return true
+	// Ensure that trailing slash is removed from the API Endpoint
+	rec.APIEndpoint.Path = strings.TrimRight(rec.APIEndpoint.Path, "/")
+
+	return rec, nil
+}
+
+func (p *portalProxy) cnsiRecordExists(endpoint string) bool {
+	log.Debug("cnsiRecordExists")
+
+	if _, err := p.GetCNSIRecordByEndpoint(endpoint); err == nil {
```

Comment:
You can remove the if:

_, err := p.GetCNSIRecordByEndpoint(endpoint)
return err == nil